File: bindings/csharp/Candle.cs
namespace OpenLimits
{
    using System.Runtime.InteropServices;
    [StructLayout(LayoutKind.Sequential, Pack = 1)]
    public struct Candle
    {
        public readonly ulong time;
        public readonly double low;
        public readonly double high;
        public readonly double open;
        public readonly double close;
        public readonly double volume;

        public Candle(ulong time, double low, double high, double open, double close, double volume)
        {
            this.time = time;
            this.low = low;
            this.high = high;
            this.open = open;
            this.close = close;
            this.volume = volume;
        }
    }
}

File: crates/openlimits-coinbase/src/model/candle.rs
use serde::Deserialize;
use serde::Serialize;
use rust_decimal::prelude::Decimal;

/// This struct represents the candle
#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct Candle {
    pub time: u64,
    pub low: Decimal,
    pub high: Decimal,
    pub open: Decimal,
    pub close: Decimal,
    pub volume: Decimal,
}